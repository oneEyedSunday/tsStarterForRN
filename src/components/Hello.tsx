import * as React from 'react'
import {Button, Text, View, StyleSheet} from 'react-native'

export interface Props {}

interface State {}


export default class Hello extends React.Component<Props, State> {
    constructor(props: Props){
        super(props)
        // set state
    }

    // methods


    render(){
        return (
            <View style={styles.container}>
                <Text style={styles.fancyLiturgy}>Hello from the other side</Text>
            </View>
        )
    }
   
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: 'yellow',
        alignItems: 'center',
        justifyContent: 'space-around'
    },
    fancyLiturgy: {
        fontSize: 24,
        fontStyle: 'italic',
        fontFamily: " 'Century Gothic', monospace sans-serif"
    }
})