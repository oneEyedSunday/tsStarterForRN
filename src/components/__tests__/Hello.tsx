import React from 'react'
import renderer from 'react-test-renderer'

import Hello from '../Hello'

it("renders correctly with defaults", () => {
    const button = renderer.create(<Hello />).toJSON()
    expect(button).toMatchSnapshot()
})