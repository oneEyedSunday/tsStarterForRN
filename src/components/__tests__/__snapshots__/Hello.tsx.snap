// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "yellow",
      "flex": 1,
      "justifyContent": "space-around",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontFamily": " 'Century Gothic', monospace sans-serif",
        "fontSize": 24,
        "fontStyle": "italic",
      }
    }
  >
    Hello from the other side
  </Text>
</View>
`;
